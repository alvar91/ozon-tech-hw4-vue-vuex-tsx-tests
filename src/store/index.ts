import Vue from "vue";
import Vuex from 'vuex';
import { createVuexStore } from "vuex-simple";

// @ts-ignore
import { MainModule } from "./modules/main/main.ts";

Vue.use(Vuex);

const store = createVuexStore(new MainModule());

export default store;
