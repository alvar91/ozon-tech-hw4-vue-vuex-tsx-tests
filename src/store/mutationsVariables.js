export const FETCH_USERS_ACTION = "FETCH_USERS_ACTION";
export const FETCH_USERS = "FETCH_USERS";
export const SET_LOADING = "SET_LOADING";
export const SET_CURRENT_USER = "SET_CURRENT_USER";
export const SET_FILTER_KEYWORD = "SET_FILTER_KEYWORD";
export const SET_FORM_INPUT = "SET_FORM_INPUT";
export const SET_FORM_CHANGE = "SET_FORM_CHANGE";
export const SET_FORM_RADIO = "SET_FORM_RADIO";
