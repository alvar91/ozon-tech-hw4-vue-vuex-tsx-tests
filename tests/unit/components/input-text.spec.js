import Vuex from "vuex";
import { mount, createLocalVue } from "@vue/test-utils";

import InputText from "@/components/input-text.vue";

import { mockUserName, mockUserId, mockUser } from "../../mock/mockUser";

import {
  FETCH_USERS_ACTION,
  SET_CURRENT_USER,
} from "@/store/mutationsVariables";

import store from "@/store/index";

const localVue = createLocalVue();
localVue.use(Vuex);

const { middleName } = mockUser;
describe("InputText.vue", () => {
  let wrapper;

  beforeEach(async () => {
    await store.dispatch(FETCH_USERS_ACTION);
    store.commit(SET_CURRENT_USER, mockUserId);
  });

  it("should correctly renders markup with default props", () => {
    wrapper = mount(InputText, {
      localVue,
      propsData: {
        title: "Фамилия",
        name: "middleName",
        value: { middleName },
        placeholder: "Фамилия",
        isDisabled: false,
        isWide: false,
      },
      store,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should correctly renders markup with wided input", () => {
    wrapper = mount(InputText, {
      localVue,
      propsData: {
        title: "Фамилия",
        name: "middleName",
        value: { middleName },
        placeholder: "Фамилия",
        isDisabled: false,
        isWide: true,
      },
      store,
    });

    expect(wrapper.find(".card__field--wide").exists()).toBeTruthy();
  });

  it("should correctly renders markup and don`t change the store with disabled input", () => {
    wrapper = mount(InputText, {
      localVue,
      propsData: {
        title: "Фамилия",
        name: "middleName",
        value: { middleName },
        placeholder: "Фамилия",
        isDisabled: true,
        isWide: false,
      },
      store,
    });

    expect(wrapper.find(".card__field--disabled").exists()).toBeTruthy();

    const input = wrapper.find(".card__input");
    input.element.value = mockUserName;
    input.trigger("input");

    expect(store.getters.currentUser.middleName).not.toBe(mockUserName);
  });

  it("should correctly input data and change the store", () => {
    wrapper = mount(InputText, {
      localVue,
      propsData: {
        title: "Фамилия",
        name: "middleName",
        value: { middleName },
        placeholder: "Фамилия",
        isDisabled: false,
        isWide: false,
      },
      store,
    });

    const input = wrapper.find(".card__input");
    input.element.value = mockUserName;
    input.trigger("input");

    expect(store.getters.currentUser.middleName).toBe(mockUserName);
  });

  afterEach(() => {
    wrapper.destroy();
  });
});
