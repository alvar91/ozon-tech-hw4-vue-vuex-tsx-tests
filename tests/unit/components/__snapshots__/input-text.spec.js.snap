// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText.vue should correctly renders markup with default props 1`] = `<label class="card__field"><span class="card__title">Фамилия</span><input type="text" name="middleName" placeholder="Фамилия" class="card__input"></label>`;
